modules/ROOT/pages/salesforce/salesforce-connector-96.adoc

```diff
@@ -0,0 +1,1211 @@
+
+Each broadcast event is assigned a numeric ID. IDs are incremented and not guaranteed to be contiguous for consecutive events. Each ID is guaranteed to be higher than the ID of the previous event. For example, the event following the event with ID 999 can have an ID of 1025. The ID is unique for the organization and the channel. The IDs of deleted events are not reused.
+
+See also <<Receive Inbound Data From Salesforce>> for event processing
+when streaming data to an application from Salesforce.
+
+Topics:
+
+* <<Subscribe to a Topic>>
+* <<Receive Events for a Topic>>
+* <<Receive Custom Event Notifications>>
+* <<Replay Events From a Topic>>
+
+=== Subscribe to a Topic
+
+After you create a topic, you can start receiving events by subscribing to the topic. To do so, you add the Subscribe Topic (`subscribe-topic`) or a Replay Topic (`replay-topic`) trigger to your flow. The trigger acts as an inbound endpoint. Every time the subscription receives an event, the trigger executes the rest of the flow in your Mule app. In the case of the XML example below, it prints a message to the log at INFO level.
+
+In Design Center, you use Subscribe Topic or Replay Topic operations for the Salesforce connector as the trigger.
+
+In XML, you use `subscribe-topic` or `replay-topic` as the trigger:
+
+[source,xml,linenums]
+----
+<flow name="accountUpdatesSubscription">
+    <!-- INBOUND ENDPOINT -->
+    <sfdc:subscribe-topic topic="AccountUpdates"/>
+    <!-- REST OF YOUR FLOW -->
+    <logger level="INFO" message="Received an event for Salesforce Object ID #[map-payload:Id]"/>
+</flow>
+----
+
+*Note:* When subscribing to a topic that was not previously published in Salesforce, the subscription is successful. When the topic is later published, the user who is already subscribed to it does NOT receive notifications regarding that topic. The user has to resubscribe after the topic creates.
+
+=== Receive Events for a Topic
+
+Before you can receive events for Salesforce changes that are associated with a topic, you must first create a topic (a PushTopic). A PushTopic is a special object in Salesforce that binds a name (the topic's name) and Salesforce Object Query Language (SOQL) query together. Once a PushTopic is created, you can subscribe to it by using its name.
+
+In Design Center, you can either use the Create (`create`) or Publish Topic (`publish-topic`) operation to create a topic. Example of the required fields for these operations:
+
+* Topic Name: `AccountUpdates`
+* Query: `SELECT Id, Name FROM Account`
+
+Example in XML for `publish-topic`:
+
+`<sfdc:publish-topic name="AccountUpdates" query="SELECT Id, Name FROM Account"/>`
+
+Alternatively, in Salesforce you might create a topic by executing code like this from an Enter Apex Code window, accessible through your system logs:
```

Comment:
Huh? Are you referencing the Developer Console? How would I get to that window from "system logs"?